packages/gatsby/cache-dir/develop-static-entry.js

```diff
@@ -1,7 +1,14 @@
 import React from "react"
-import { renderToStaticMarkup } from "react-dom/server"
+import fs from "fs"
+import { renderToString, renderToStaticMarkup } from "react-dom/server"
 import { merge } from "lodash"
+import { join } from "path"
 import apiRunner from "./api-runner-ssr"
+import { grabMatchParams } from "./find-path"
+import syncRequires from "$virtual/sync-requires"
```

Comment:
@wardpeet asked about the memory implications of loading in all the page components.

I created a test site with 1000 page components (so indicative of a site with a reasonably large amount of JavaScript).

I measured the heap size after the dev started on with both master & this PR and... it was almost identical (within 10mb of each other (runs varied quite a bit too — so "within standard of error" I suppose). Which was a bit surprising but good.

More concerning is this does increase the JS processing time. This PR increased cold starts from ~42s to 70s. Warm starts for master is ~15s vs ~23s for this PR so somewhat smaller delta.

Reply:
The only way around this I've thought of is to lazily add page components to the sync-requires file — start w/ none & then when a page is requested add it & wait for webpack to compile and then render the page.

Reply:
Holy crap, that works... commit coming

Reply:
With lazy adding page components — cold builds are only 4 seconds slower (42-46) and warm builds are only 1 second slower (15-16s)

Reply:
The initial page render is ~500ms slower which is the time it takes to recreate the SSR bundle (that number will vary of course depending on the size of the page component — but it's a one-time cost per dev session).